core/src/util/constants.rs

```diff
@@ -16,3 +16,6 @@
 
 /// The target where to log check-if-email-exists logs.
 pub const LOG_TARGET: &str = "reacher";
+pub const HAVEIBEENPWNED_ENDPOINT: &str = "https://haveibeenpwned.com/unifiedsearch/";
```

Comment:
Sorry I didn't reply in the original issue, but I would prefer to stick with the paid API for now, and not use headless browsers.

I think they might work today, but there's a high probability they will become flaky tomorrow. The maintenance overhead for me is too big compared to the benefit IMO.